beamer/health/check.py

```diff
@@ -0,0 +1,526 @@
+
+    return liquidity_text
+
+
+def create_unclaimed_fill_message(request: RequestCreated, fill: RequestFilled) -> Notification:
+    return {
+        "meta": {
+            "request_id": request.request_id.hex(),
+            "message_type": MessageTypes.UNCLAIMED_FILL,
+            "message_link": link_to_explorer(request.chain_id, request.tx_hash.hex()),
+        },
+        "body": f"""
+v1: Unclaimed fill!
+Request: `{request.request_id.hex()}`
+Value: `{request.amount}` | `{request.amount * 1e-6}` USDC
+Block: `{request.block_number}`
+Filler: `{fill.filler}`
+TX_Hash `{fill.tx_hash.hex()}`
+Valid_until: {request.valid_until} | `{datetime.fromtimestamp(request.valid_until)
+        .strftime("%d.%m.%Y %H:%M:%S")}`
+        """,
+    }
+
+
+def create_expired_request_message(request: RequestCreated) -> Notification:
+    return {
+        "meta": {
+            "request_id": request.request_id.hex(),
+            "message_type": MessageTypes.REQUEST_EXPIRED,
+            "message_link": link_to_explorer(request.chain_id, request.tx_hash.hex()),
+        },
+        "body": f"""
+Request expired with no fill {request.request_id.hex()}
+Request: `{request.request_id.hex()}`
+Value: `{request.amount}` | `{request.amount * 1e-6}` USDC
+Valid_until: {request.valid_until} | `{datetime.fromtimestamp(request.valid_until)
+        .strftime("%d.%m.%Y %H:%M:%S")}`
+TX_Hash `{request.tx_hash.hex()}`
+        """,
+    }
+
+
+def create_challenge_game_message(
+    message_type: str, request: RequestCreated, fill: RequestFilled, last_claim: ClaimMade
+) -> Notification:
+    title = {
+        MessageTypes.CHALLENGE_GAME: "Challenge game!",
+        MessageTypes.CHALLENGE_GAME_CLAIM_BY_SOMEONE_ELSE: "Claimed by someone else!",
+    }
+
+    return {
+        "meta": {
+            "request_id": request.request_id.hex(),
+            "message_type": message_type,
+            "message_link": link_to_explorer(request.chain_id, last_claim.tx_hash.hex()),
+        },
+        "body": f"""
+v1: {title[message_type]}
+Request: `{request.request_id.hex()}`
+Claim_id: `{last_claim.claim_id}`
+Request filled by: `{fill.filler}` | `{fill.tx_hash.hex()}`
+Claimer: `{last_claim.claimer}`
+Last_challenger: `{last_claim.last_challenger}`
+Fill_id: `{fill.fill_id.hex()}`
+Claimer_stake: `{last_claim.claimer_stake}` | `{Web3.fromWei(last_claim.claimer_stake, 'ether')}` ethers
+Challenger_stake_total: `{last_claim.challenger_stake_total}` | `{Web3.fromWei(last_claim.challenger_stake_total, 'ether')}` ETH
+Block: `{last_claim.block_number}`
+Terminates: `{last_claim.termination}` | `{datetime.fromtimestamp(last_claim.termination).strftime("%d.%m.%Y %H:%M:%S")}`
+TX_Hash `{last_claim.tx_hash.hex()}`
+       """,  # noqa: E501
+    }
+
+
+def process_messages(ctx: Context) -> None:
+    notify = Notify(get_config()["notification_system"], get_config()["notification"])
+    for message in ctx.messages:
+
+        def callback(mess: Notification = message) -> None:
```

Comment:
This looks strange. `mess` is supposed to be a message, yet the type is `Notification`?

Reply:
Hm, maybe I have some kind of mix up in the whole class. 

In my head it all comes from Notification message / Push Notification. 

On iOS/Android a push notification is a data object sent to a device. This object contains params such as:

title
text
but also might contain meta information such as -> badge_icon, sound, content_id etc. 

So originally I started up with a message - which in the context of rocketchat is just a text, but then we needed some more meta information and I went with Notification. So now the callback expects a notification, because writing what has been sent needs the request_id,, but the actual sending needs the text :D

Another funny thing that I realize right now is that on my context I do
`ctx.add_message` and then call one of `create...message` functions, but they all actually return a `Notification`, so a more correct naming would be: `add_notification`, `create...notification`